nixos/modules/services/mail/listmonk.nix

```diff
@@ -0,0 +1,212 @@
+            };
+
+            "privacy.exportable" = mkOption {
+              type = listOf str;
+              default = ["profile" "subscriptions" "campaign_views" "link_clicks"];
+              description = lib.mdDoc "List of fields which can be exported through an automatic export request";
+            };
+
+            "privacy.domain_blocklist" = mkOption {
+              type = listOf str;
+              default = [];
+              description = lib.mdDoc "E-mail addresses with these domains are disallowed from subscribing.";
+            };
+
+            smtp = mkOption {
+              type = listOf (submodule {
+                freeformType = with types; attrsOf (oneOf [ str int bool ]);
+
+                options = {
+                  enabled = mkEnableOption "this SMTP server for listmonk";
+                  host = mkOption {
+                    type = types.str;
+                    description = lib.mdDoc "Hostname for the SMTP server";
+                  };
+                  port = mkOption {
+                    type = types.port;
+                    description = lib.mdDoc "Port for the SMTP server";
+                  };
+                  max_conns = mkOption {
+                    type = types.int;
+                    description = lib.mdDoc "Maximum number of simultaneous connections, defaults to 1";
+                    default = 1;
+                  };
+                  tls_type = mkOption {
+                    type = types.enum [ "none" "STARTTLS" "TLS" ];
+                    description = lib.mdDoc "Type of TLS authentication with the SMTP server";
+                  };
+                };
+              });
+
+              description = lib.mdDoc "List of outgoing SMTP servers";
+            };
+
+            # TODO: refine this type based on the smtp one.
+            "bounce.mailboxes" = mkOption {
+              type = listOf (submodule {
+                freeformType = with types; oneOf [ str int bool ];
+              });
+              default = [];
+              description = lib.mdDoc "List of bounce mailboxes";
+            };
+
+            messengers = mkOption {
+              type = listOf str;
+              default = [];
+              description = lib.mdDoc "List of messengers, see: <https://github.com/knadh/listmonk/blob/master/models/settings.go#L64-L74> for options.";
+            };
+          };
+        });
+
+        description = lib.mdDoc "Dynamic settings in the PostgreSQL database, set by a SQL script, see <https://github.com/knadh/listmonk/blob/master/schema.sql#L177-L230> for details.";
+      };
+      secretFile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = lib.mdDoc "Secret file loaded as an environment file, read <https://listmonk.app/docs/configuration/#environment-variables> for details.";
+      };
+    };
+  };
+
+
+  ###### implementation
+  config = mkIf cfg.enable (mkMerge [
```

Comment:
Feel free to completely ignore this comment because it is preference only but I prefer to avoid a large `mkMerge` like this in favour of `mkMerge` only applied to specific options as needed - as deep as possible within the option stack.

Reply:
Addressed.